sdks/java/core/src/test/java/org/apache/beam/sdk/io/ReadTest.java

```diff
@@ -17,28 +17,35 @@
  */
 package org.apache.beam.sdk.io;
 
+import static org.apache.beam.sdk.transforms.display.DisplayDataMatchers.hasDisplayItem;
+import static org.apache.beam.sdk.transforms.display.DisplayDataMatchers.includes;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+
 import org.apache.beam.sdk.coders.Coder;
 import org.apache.beam.sdk.io.UnboundedSource.CheckpointMark;
 import org.apache.beam.sdk.options.PipelineOptions;
+import org.apache.beam.sdk.transforms.display.DisplayData;
 
+import org.joda.time.Duration;
 import org.junit.Rule;
 import org.junit.Test;
 import org.junit.rules.ExpectedException;
 import org.junit.runner.RunWith;
 import org.junit.runners.JUnit4;
 
 import java.io.IOException;
+import java.io.Serializable;
 import java.util.List;
-
 import javax.annotation.Nullable;
 
 /**
  * Tests for {@link Read}.
  */
 @RunWith(JUnit4.class)
-public class ReadTest {
+public class ReadTest implements Serializable{
   @Rule
-  public ExpectedException thrown = ExpectedException.none();
+  public transient ExpectedException thrown = ExpectedException.none();
```

Comment:
I'm going to leave it. Anonymous classes end up being useful in many test cases. I could move this case but it seems likely we might want to use anonymous classes again in the future.